File: .skills/openclaw-skills/skills/hfarazi/ticktickpower/ticktick/commands/list.py
import json
import sys

from ..api import api


def _normalize_color(color: str) -> str:
    return color if color.startswith("#") else f"#{color}"


def list_create_command(args) -> None:
    try:
        color = _normalize_color(args.color) if args.color else None
        project = api.create_project(args.name, color)

        if args.json:
            print(json.dumps(project, indent=2))
            return

        print(f'✓ Project created: "{project["name"]}"')
        print(f'  ID: {project["id"]}')
        if project.get("color"):
            print(f'  Color: {project["color"]}')

    except RuntimeError as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)


def list_update_command(args) -> None:
    try:
        project = api.find_project_by_name(args.name)
        if not project:
            print(f"Project not found: {args.name}", file=sys.stderr)
            sys.exit(1)

        new_name = args.new_name if args.new_name else None
        color = _normalize_color(args.color) if args.color else None

        updated = api.update_project(project["id"], name=new_name, color=color)

        if args.json:
            print(json.dumps(updated, indent=2))
            return

        print(f'✓ Project updated: "{updated.get("name", project["name"])}"')
        print(f'  ID: {updated.get("id", project["id"])}')
        if updated.get("color"):
            print(f'  Color: {updated["color"]}')

    except RuntimeError as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)

File: .skills/openclaw-skills/skills/norickkevorkov/ticktick-cli/scripts/commands/list.ts
import { api, type CreateProjectInput, type UpdateProjectInput } from "../api";

interface ListCreateOptions {
  color?: string;
  json?: boolean;
}

interface ListUpdateOptions {
  update: boolean;
  name?: string;
  color?: string;
  json?: boolean;
}

function normalizeColor(color: string): string {
  // TickTick expects colors in hex format
  if (color.startsWith("#")) {
    return color;
  }
  return `#${color}`;
}

export async function listCreateCommand(
  name: string,
  options: ListCreateOptions
): Promise<void> {
  try {
    const input: CreateProjectInput = { name };

    if (options.color) {
      input.color = normalizeColor(options.color);
    }

    const project = await api.createProject(input);

    if (options.json) {
      console.log(JSON.stringify(project, null, 2));
      return;
    }

    console.log(`✓ Project created: "${project.name}"`);
    console.log(`  ID: ${project.id}`);
    if (project.color) {
      console.log(`  Color: ${project.color}`);
    }
  } catch (error) {
    console.error(
      `Error: ${error instanceof Error ? error.message : String(error)}`
    );
    process.exit(1);
  }
}

export async function listUpdateCommand(
  nameOrId: string,
  options: ListUpdateOptions
): Promise<void> {
  try {
    // Find the project
    const project = await api.findProjectByName(nameOrId);
    if (!project) {
      console.error(`Project not found: ${nameOrId}`);
      process.exit(1);
    }

    const input: UpdateProjectInput = { id: project.id };

    if (options.name) {
      input.name = options.name;
    }

    if (options.color) {
      input.color = normalizeColor(options.color);
    }

    const updated = await api.updateProject(input);

    if (options.json) {
      console.log(JSON.stringify(updated, null, 2));
      return;
    }

    console.log(`✓ Project updated: "${updated.name}"`);
    console.log(`  ID: ${updated.id}`);
    if (updated.color) {
      console.log(`  Color: ${updated.color}`);
    }
  } catch (error) {
    console.error(
      `Error: ${error instanceof Error ? error.message : String(error)}`
    );
    process.exit(1);
  }
}
